src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolWorkQueue.cs

```diff
@@ -479,7 +611,11 @@ public void Enqueue(object callback, bool forceGlobal)
             }
             else
             {
-                workItems.Enqueue(callback);
+                ConcurrentQueue<object> queue =
+                    AssignableWorkItemQueueCount > 0 && (tl = ThreadPoolWorkQueueThreadLocals.threadLocals) != null
+                        ? tl.assignedGlobalWorkItemQueue
+                        : workItems;
+                queue.Enqueue(callback);
```

Comment:
I assume this has zero impact on performance for <= 32 because AssignableWorkItemQueueCount will tier up as a const and this will then optimize down to just `workItems.Enqueue(callback)`.